src/Services/SearchService.php

```diff
@@ -410,26 +418,36 @@ private function getProductDetailUrl(Response $response)
             $productId = explode('_', $productId)[0];
         }
 
-        $result = $itemSearchService->getResult(
-            $this->getSearchFactory()->hasItemId($productId)
-        );
+        // Necessary to retrieve variation data and prices
+        $resultFields = $this->loadResultFieldTemplate(ResultFieldTemplate::TEMPLATE_LIST_ITEM);
```

Comment:
How do you find this? Did you check the Ceres code for this?

Reply:
Discussed in call